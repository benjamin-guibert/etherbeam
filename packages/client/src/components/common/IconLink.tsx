import React, { ReactElement, ReactNode } from 'react'
import Button from 'react-bootstrap/Button'

interface IconLinkProps {
  url: string
  title?: string
  className?: string
  children: ReactNode
}

const IconLink = ({ url, title, className, children }: IconLinkProps): ReactElement => {
  return (
    <Button
      className={['shadow-none', className].join(' ')}
      variant="link"
      title={title}
      href={url}
      onClick={(event) => event.stopPropagation()}
      target="_blank"
    >
      {children}
    </Button>
  )
}

export default IconLink
