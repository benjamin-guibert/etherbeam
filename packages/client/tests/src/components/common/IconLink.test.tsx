import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import IconLink from 'components/common/IconLink'

describe('<IconLink />', () => {
  let component: ReactWrapper

  beforeAll(() => {
    component = mount(
      <IconLink url="https://url.com" title="Title" className="app-class">
        Content
      </IconLink>
    )
  })

  it('should render', () => {
    const button = component.find('Button')
    expect(button.prop('href')).toBe('https://url.com')
    expect(button.prop('title')).toBe('Title')
    expect(button.hasClass('app-class')).toBeTruthy()
    expect(button.prop('children')).toEqual('Content')
  })

  it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
})
