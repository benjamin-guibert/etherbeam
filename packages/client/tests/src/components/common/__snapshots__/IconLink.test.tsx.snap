// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLink /> should match snapshot 1`] = `
<a
  class="shadow-none app-class btn btn-link"
  href="https://url.com"
  target="_blank"
  title="Title"
>
  Content
</a>
`;
